transform/app/services/mp_pc/data.py

```diff
@@ -68,3 +71,29 @@ async def check_mp_auth(data: dict | list[dict]) -> None:
                 raise requests.HTTPError("401 Client Error: Unauthorized")
             else:
                 raise requests.RequestException(data["error"])
+
+
+def data_source_pids_list():
+    # TODO add reset pids after full oag transform
+    """
+        Returns a list of data source PIDs, ensuring a singleton pattern to initialize it only once.
+    This function retrieves a list of data source PIDs from a data source and caches the result
+    to ensure that the initialization logic is executed only once. Subsequent calls to this
+    function will return the existing list of data source PIDs.
+    Returns:
+        list: A list of data source PIDs.
+    Example:
+        data_source_pids = data_source_pids_list()
+    """
+    if (
+        not hasattr(data_source_pids_list, "_instance")
+        or data_source_pids_list._instance is None
+    ):
+        loop = asyncio.get_event_loop()
```

Comment:
We use Python 3.10 in this project.
From Python 3.7+ it is recommended to use `asyncio.run()`

Reply:
Thanks for the review. I have addressed your recommendation to use asyncio.run(). Noted.